includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -257,21 +257,59 @@ public function sanitize_dimension( $value, $dimension ) {
 	/**
 	 * Sets the layout, and possibly the 'height' and 'width' attributes.
 	 *
-	 * @param string[] $attributes {
+	 * @param array $attributes {
 	 *      Attributes.
 	 *
-	 *      @type int $height
-	 *      @type int $width
-	 *      @type string $sizes
-	 *      @type string $class
-	 *      @type string $layout
+	 *      @type int|string $height
+	 *      @type int|string $width
+	 *      @type string     $sizes
+	 *      @type string     $class
+	 *      @type string     $layout
 	 * }
 	 * @return array Attributes.
 	 */
 	public function set_layout( $attributes ) {
 		if ( isset( $attributes['layout'] ) && ( 'fill' === $attributes['layout'] || 'flex-item' !== $attributes['layout'] ) ) {
 			return $attributes;
 		}
+
+		// Special-case handling for inline style that should be transformed into layout=fill.
+		if ( ! empty( $attributes['style'] ) ) {
+			$styles = $this->parse_style_string( $attributes['style'] );
+
+			// top, left, bottom, right are used.
```

Comment:
```suggestion
			// Apply fill layout if top, left, bottom, right are used.
```

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3209/commits/47f0f331655388c57c45c5acafc097d07330383c